src/transformers/pipelines/image_to_text.py

```diff
@@ -98,9 +105,34 @@ def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Imag
         """
         return super().__call__(images, **kwargs)
 
-    def preprocess(self, image):
+    def preprocess(self, image, prompt=None):
         image = load_image(image)
         model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+
+        if self.model.config.model_type == "git" and prompt is None:
+            model_inputs["input_ids"] = None
+
+        if prompt is not None and not isinstance(prompt, str):
+            raise ValueError(
+                f"Received an invalid text input, got - {type(prompt)} - but expected a single string. "
+                "Note also that one single text can be provided for conditional image to text generation."
+            )
+
+        elif prompt is not None and self.model.config.model_type == "git":
+            input_ids = self.tokenizer(text=prompt, add_special_tokens=False).input_ids
+            input_ids = [self.tokenizer.cls_token_id] + input_ids
+            input_ids = torch.tensor(input_ids).unsqueeze(0)
+            model_inputs.update({"input_ids": input_ids})
+
+        # vision-encoder-decoder does not support conditional generation
+        elif prompt is not None and self.model.config.model_type != "vision-encoder-decoder":
+            text_inputs = self.tokenizer(prompt, return_tensors=self.framework)
+
+            if "token_type_ids" in text_inputs:
```

Comment:
Why do we need to pop those ? What happens if we leave them ?

Reply:
This was taken from the PR here: https://github.com/huggingface/transformers/pull/22423.

However, it turns out this is very BLIP-specific. The reason is that BLIP uses `BertTokenizer`, which returns `token_type_ids` by default, but `BlipForConditionalGeneration` doesn't accept `token_type_ids`. 

I think the fix here is to include `model_input_names` in the tokenizer_config.json file of the checkpoints on the hub as was done for GIT [here](https://huggingface.co/microsoft/git-base/blob/main/tokenizer_config.json#L5-L8), and eventually remove [this line](https://github.com/huggingface/transformers/blob/c2393cad085e3875ee2206d917d46d15e50602a3/src/transformers/models/blip/processing_blip.py#L44).

@younesbelkada could you take this up?

Reply:
Thanks for the explanation, we can leave the clean up for a follow up PR.

Reply:
Sure @NielsRogge , just merged the PRs on the Hub